### PR TITLE
Fix test warnings and delayload linking

### PR DESCRIPTION
**Pull request by tiagoshibata**
**Description**: Fix test warnings w/ new test macros and small linking issues.

### CODE REVIEW

File: cmake/winml_unittests.cmake [hunk 1]
```diff
@@ -81,7 +81,7 @@ endif()
 add_winml_test(
   TARGET winml_test_scenario
   SOURCES ${winml_test_scenario_src}
-  LIBS winml_test_common ${winml_test_scenario_libs}
+  LIBS winml_test_common delayimp.lib ${winml_test_scenario_libs}
```
Comment (tiagoshibata) [inline cmake/winml_unittests.cmake]: Needed because of line 89, which sets delay loaded libs

File: winml/test/common/test.h [hunk 2]
```diff
@@ -1,10 +1,18 @@
+#pragma once
+
 using VoidTest = void (*)();
 using SetupTest = VoidTest;
 
+constexpr bool alwaysTrue() {
+    return true;
+}
+#define WINML_SUPRESS_UNREACHABLE_BELOW(statement)    \
+    if (alwaysTrue()) { statement; }
```
Comment (tiagoshibata) [inline winml/test/common/test.h]: I tried if constexpr which would ensure the unused code doesn't get compiled, but VS still shows a warning (looks like a bug: https://developercommunity.visualstudio.com/content/problem/421660/warning-c4702-is-overzealous-with-if-constexpr-gua.html)

Comment (snnn) [reply winml/test/common/test.h]: Is it required to fix C4702 warnings ?

Comment (tiagoshibata) [reply winml/test/common/test.h]: Yes, the code after GPUTEST is unreachable if building without DML

**Pull request merged**